### ISSUE TITLE
Reactive conditions

### ISSUE THREAD

**Issue by kethan** [Technical-Source/bruh#3]
Hi, very nice library.

How do I under such conditions in the counter example?

```js
{count.value % 2 === 0 ? <div>Even</div>: <div>Odd</div>}
```

**Comment by kethan** [comment]
@wlib

**Comment by wlib** [comment]
Sorry about the late reply Kethan. Bruh is a bit unlike react in some ways; that `count.value` access would only run once (because it just gets the _current_ value of `count`). To get the desired behavior, the "Odd"/"Even" should be a reactive value itself.

```jsx
const parity = r([count], () =>
  count.value % 2 === 0
    ? "Even"
    : "Odd"
)

return <div>{ parity }</div>
```

**Comment by kethan** [comment]
Oh wow that's so simple. Before I tried with https://github.com/kethan/ulive instead of your reactivity it works like charm.

**Comment by kethan** [comment]
@wlib is there a way I can implement a clean up function when component exits like i want to remove event listeners, etc ??

**Comment by wlib** [comment]
I'm not quite sure how to answer that because there is no component lifecycle, so there is no component unmounting. This may change in the future, but I'm trying to keep it as close to "standard html/js" as I can for future-proofing. The trouble with lifecycle methods is that they would force bruh to constrain how you use elements a little too much, as bruh's elements are nothing more than plain elements created with `document.createElement()`. It's too difficult to tell when an element is really "done" automatically because you could always just take an element out of the DOM and place it somewhere else anytime later.

The HTML standard has moved towards [`AbortSignal`](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) for some things like `fetch()` cancellation and [custom cancellation](https://dom.spec.whatwg.org/#aborting-ongoing-activities-example). It's neat because one signal can be used for everything that should be affected, as shown [here for multiple fetches](https://developer.chrome.com/blog/abortable-fetch/#one-signal-many-fetches) and [here for multiple event listeners](https://css-tricks.com/using-abortcontroller-as-an-alternative-for-removing-event-listeners/).

If you should me your exact use case I might be able to show you how I would do it. If I don't have a good solution right now, then it's something I will definitely need to think about how to add support for.

**Comment by kethan** [comment]
Yes you right. But i do have an usecase i will try to share with you. It's just like solidjs oncleanup function it just cleans up old effects when run new effects.
